sensor/common/detector/detector.go

```diff
@@ -274,6 +274,13 @@ func (d *detectorImpl) processUpdatedImage(image *storage.Image) error {
 	}
 	d.enricher.imageCache.Add(key, newValue)
 	d.admissionCacheNeedsFlush = true
+	d.deduper.reset()
+	// Take deployment lock and flush
+	concurrency.WithLock(&d.deploymentDetectionLock, func() {
+		for _, deployment := range d.deploymentStore.GetAll() {
```

Comment:
If we are in the middle of processing a deployment we might send a deployment (i.e. in between handler and resolver) to detector with incorrect permission levels or exposure info. The alerts will be fixed right after since there is a deployment in pipeline, but still, some incorrect alerts will be sent.